doc/stdenv/stdenv.chapter.md

```diff
@@ -206,6 +206,15 @@ For example, software using Wayland usually needs the `wayland` library at runti
 But it also executes the `wayland-scanner` program as part of the build to generate code, so `wayland` should also be added to `nativeBuildInputs`.
 :::
 
+::: {.note}
+When using native compilation, `stdenv` is lenient towards incorrect uses of `buildInputs` and `nativeBuildInputs`.
```

Comment:
May not be necessary:

1. Mention this behavior can break cross compilation
2. Move introduction of `strictDeps` to a section?